packages/paste-core/components/code-block/src/CodeBlockHeader.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import type {BoxProps} from '@twilio-paste/box';
+import type {asTags} from '@twilio-paste/heading';
+
+export interface CodeBlockHeaderProps extends Partial<Omit<HTMLHeadingElement, 'children'>> {
+  children: string;
+  element?: BoxProps['element'];
+  as?: asTags;
+}
+
+export const CodeBlockHeader = React.forwardRef<HTMLHeadingElement, CodeBlockHeaderProps>(
+  ({children, element = 'CODE_BLOCK_HEADER', as = 'h3', ...props}, ref) => (
+    <Box
+      {...safelySpreadBoxProps(props)}
+      ref={ref}
+      element={element}
+      as={as}
```

Comment:
question: passing h1/h2/h3/etc doesn't change the CodeBlockHeader visually, so why do we allow for the different options? The docs say "use the correct heading level" to ensure a11y but in what situation would one be correct over another? Do we really want people to have h1 code block headers?

Reply:
It all depends on the heading hierarchy of the page. I doubt there would ever be an H1 code block header, but I can totally believe other heading levels - it all depends on the other headers on the page. For example, if the code block is underneath an H3, then the code block heading should be an h4. 


You can read more about heading levels here: https://www.w3.org/WAI/tutorials/page-structure/headings/

Reply:
totally non-blocking but might be helpful to add a bit of this context to the a11y section of the docs page